sequencer/src/header.rs

```diff
@@ -191,16 +191,21 @@ impl Header {
 
         // fetch receipts from the l1
         let receipts = fetch_fee_receipts(parent.l1_finalized, l1.finalized);
-        for FeeReceipt { recipient, amount } in receipts {
+        for receipt in receipts {
+            let account = receipt.account();
+            let amount = receipt.amount();
+
             // Get the balance in order to add amount, ignoring the proof.
-            match fee_merkle_tree.universal_lookup(recipient) {
-                LookupResult::Ok(balance, _) => fee_merkle_tree
-                    .update(recipient, balance.add(amount))
-                    .unwrap(),
+            match fee_merkle_tree.update_with(account, |balance| {
+                Some(balance.cloned().unwrap_or_default().add(amount))
+            }) {
+                Ok(LookupResult::Ok(..)) => (),
                 // Handle `NotFound` and `NotInMemory` by initializing
```

Comment:
`NotFound` is now handled by `update_with` (the case where we `unwrap_or_default` to initialize the balance. `NotInMemory` will be handled in future work by catchup